package/kedro_viz/server.py

```diff
@@ -32,11 +33,67 @@ def nodes():
                 "outputs": [ds.split("@")[0] for ds in n.outputs],
                 "tags": list(n.tags),
             }
-            for n in pipeline.nodes
+            for n in sorted(pipeline.nodes)
         ]
     )
 
 
+@app.route("/api/nodes.json")
+def nodes_json():
+    """Serve the pipeline data."""
+
+    def pretty_name(name):
+        name = name.replace("-", " ").replace("_", " ")
+        parts = [n[0].upper() + n[1:] for n in name.split()]
+        return " ".join(parts)
+
+    pipeline = get_project_context("create_pipeline")()
```

Comment:
What would happen if the passed pipeline is broken? Would it fail somewhere inside this function?

Reply:
it's up to the Pipeline constructor to ensure the constructed pipeline is valid